src/content/en/updates/2017/09/sensors-for-the-web.md

```diff
@@ -0,0 +1,410 @@
+modify rotation quaternion of a 3D model. The <code>model</code> is a three.js
+<code>[Object3D](https://threejs.org/docs/index.html#api/core/Object3D)</code> class instance
+that has <code>[quaternion](https://threejs.org/docs/index.html#api/core/Object3D.quaternion)</code>
+property. The following code snippet from the
+ [orientation phone](https://github.com/intel/generic-sensor-demos/tree/master/orientation-phone)
+demo, illustrates how absolute orientation sensor can be used to rotate 3D model.
+
+    function initSensor() {
+        sensor = new AbsoluteOrientationSensor({frequency: 60});
+        sensor.onreading = () => model.quaternion.fromArray(sensor.quaternion);
+        sensor.onerror = event => {
+            if (event.error.name == 'NotReadableError') {
+                console.log("Sensor is not available.");
+            }
+        }
+        sensor.start();
+    }
+
+Device's orientation will be reflected in 3D <code>model</code> rotation within the WebGL scene.
+
+<figure align="center">
+  <img src="/web/updates/images/2017/09/sensors/orientation_phone.png"
+    alt="Sensor updates 3D model's orientation">
+  <figcaption><b>Figure 4</b>: Sensor updates orientation of a 3D model</figcaption>
+</figure>
+
+### Punchmeter
+
+The following code snippet is extracted from the
+[punchmeter demo](https://github.com/intel/generic-sensor-demos/tree/master/punchmeter),
+illustrating how linear acceleration sensor can be used to calculate the maximum
+velocity of a device under the assumption that it is initially laying still.
+
+```
+   this.maxSpeed = 0;
+   this.vx = 0;
+
+   this.ax = 0;
+   this.t = 0;
+
+   function onreading() {
+     let dt = (this.accel.timestamp - this.t) * 0.001; // In seconds.
+     this.vx += (this.accel.x + this.ax) / 2 * dt;
+
+     let speed = Math.abs(this.vx);
+
+     if (this.maxSpeed < speed)
+       this.maxSpeed = speed;
+
+     this.t = this.accel.timestamp;
+     this.ax = this.accel.x;
+   }
+   ....
+
+   this.accel.addEventListener('reading', onreading);
+
+```
+
+The current velocity is calculated as an approximation to integral of the
+acceleration function.
+
+
+<figure align="center">
+  <img src="/web/updates/images/2017/09/sensors/punchmeter.png"
+    alt="Demo web application for punch speed measurument">
+  <figcaption><b>Figure 5</b>: Measurement of a punch speed</figcaption>
+</figure>
+
+
+## Privacy and Security {: privacy-and-security }
+
+Sensor readings are sensitive data which can be subject to various attacks from
+malicious web pages. Chrome implementation of Generic Sensor APIs embarked on
+few limitations to mitigate the possible security and privacy risks. These
+limitations must be taken into account by developers who intend to use the API,
+so let’s briefly enumerate them.
+
+### Only HTTPS
+
+Because Generic Sensor API is a powerful new feature added to the Web, Chrome
+aims to make it available only to secure contexts. In practice it means that
```

Comment:
The phrase 'aims to' makes it sound like we don't always success. Use: 'Chrome only allows it on secure contexts.'